src/gui/components/wxGameList.cpp

```diff
@@ -142,6 +143,60 @@ void wxGameList::LoadConfig()
 	}
 }
 
+// for unknow reasons of wxWidgets, there are some issues if we change column size in a SizeEvent handler func when use sizer-based layout.
+// the list may not redraw correctly, or extra blank rows/columns shown in list...
+// So we DO NOT change column size here, just send a ColumnResize event to ColumnResize handler, resize will be done over there.
+void wxGameList::OnGameListSize(wxSizeEvent &event)
+{
+	event.Skip();
+
+	int last_col_index = GetColumnIndexFromOrder(GetColumnCount() - 1);
+	wxListEvent column_resize_event(wxEVT_LIST_COL_END_DRAG);
+	column_resize_event.SetColumn(last_col_index);
+	wxPostEvent(this, column_resize_event);
+}
+
+// adjust the width of the last column to fit the window, whether or not the column order has been changed
+void wxGameList::AdjustLastColumnWidth()
+{
+	wxWindowUpdateLocker windowlock(this);
+	int last_col_index = GetColumnIndexFromOrder(GetColumnCount() - 1);
+	int last_col_width = GetClientSize().GetWidth();
+	for (int i = 1; i < GetColumnCount(); i++)
+	{
+		if (i != last_col_index)
+			last_col_width -= GetColumnWidth(i);
+	}
+	if (last_col_width < GetColumnDefaultWidth(last_col_index)) // keep a minimum width
+		last_col_width = GetColumnDefaultWidth(last_col_index);
+	SetColumnWidth(last_col_index, last_col_width);
+}
+
+// it's not elegant for use this GetColumnDefaultWidth() here, but it does work at the moment before we have better solution
+// return default width of column by index col
```

Comment:
Think this comment is a little long for what is really just a todo comment.
```suggestion
// todo: scale all columns using a ratio instead of hardcoding exact widths
```
Also, the function name is pretty clear, don't need to describe what it does.